sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileDataSourceV2.scala

```diff
@@ -112,4 +112,21 @@ trait FileDataSourceV2 extends TableProvider with DataSourceRegister {
       getTable(new CaseInsensitiveStringMap(properties), schema)
     }
   }
+
+  /**
+   * Check whether target schema field name is valid.
+   *
+   * @throws AnalysisException If field name is invalid
+   */
+  def checkFieldName(name: String): Unit = {}
+
+  def checkFieldNames(schema: StructType): Unit = {
+    schema.foreach { field =>
+      checkFieldName(field.name)
+      field.dataType match {
+        case s: StructType => checkFieldNames(s)
+        case _ =>
+      }
```

Comment:
I don't have a preference. Just one thought:

```suggestion
      if (field.dataType.isInstanceOf[StructType]) checkFieldNames(field.dataType.asInstanceOf[StructType])
```

up to your preference